tools/bin/hawqregister

```diff
@@ -126,182 +127,319 @@ def option_parser_yml(yml_file):
+        self.accessor = GpRegisterAccessor(conn)
+        self.utility_accessor = GpRegisterAccessor(utility_conn)
+        self.mode = self._init_mode(options.force, options.repair)
+        self._init()
+
+    def _init_mode(self, force, repair):
+        def table_existed():
+            return self.accessor.get_table_existed(self.tablename)
+
+        if self.yml:
+            if force:
+                return 'force'
+            elif repair:
+                if not table_existed():
+                    logger.error('--repair mode asserts the table is already create.')
+                    sys.exit(1)
+                return 'repair'
+            else:
+                return 'second_normal'
+        else:
+            return 'first'
 
-    return relname
+    def _init(self):
+        def check_hash_type():
+            self.accessor.check_hash_type(self.tablename)
 
+        # check conflicting distributed policy
+        def check_distribution_policy():
+            if self.distribution_policy.startswith('DISTRIBUTED BY'):
+                if len(self.files) % self.bucket_number != 0:
+                    logger.error('Files to be registered must be multiple times to the bucket number of hash table.')
+                    sys.exit(1)
 
-def check_hash_type(dburl, tablename):
-    '''Check whether target table is hash distributed, in that case simple insertion does not work'''
-    try:
-        query = "select attrnums from gp_distribution_policy, pg_class where pg_class.relname = '%s' and pg_class.oid = gp_distribution_policy.localoid;" % tablename
-        conn = dbconn.connect(dburl, False)
-        rows = dbconn.execSQL(conn, query)
-        conn.commit()
-        if not rows.rowcount:
-            logger.error('Table %s not found in table gp_distribution_policy.' % tablename)
-            sys.exit(1)
-        for row in rows:
-            if row[0]:
-                logger.error('Cannot register file(s) to a table which is hash distribuetd.')
+        def create_table():
+            return self.accessor.do_create_table(self.tablename, self.schema, self.file_format, self.distribution_policy, self.file_locations, self.bucket_number)
+
+        def get_seg_name():
+            return self.utility_accessor.get_seg_name(self.tablename, self.database, self.file_format)
+
+        def get_metadata():
+            return self.accessor.get_metadata_from_database(self.tablename, self.seg_name)
+
+        def get_distribution_policy():
+            return self.accessor.get_distribution_policy_info(self.tablename)
+
+        def check_policy_consistency():
+            policy = get_distribution_policy() # "" or "{1,3}"
+            if policy is None:
+                if ' '.join(self.distribution_policy.strip().split()).lower() == 'distributed randomly':
+                    return
+                else:
+                    logger.error('Distribution policy of %s is not consistent with previous policy.' % self.tablename)
```

Comment:
What's previous policy? Could we change "previous" to a more specific description?